### PR TITLE
Adding to Transmogrifier to support Polyhedral Meshes

### PR DESCRIPTION
**Pull request by JustinPrivitera**
Note: This requires an updated conduit (for the new generate_sides method)

### CODE REVIEW

File: src/ascent/runtimes/ascent_transmogrifier.cpp [hunk 1]
```diff
@@ -107,6 +109,147 @@ conduit::Node* Transmogrifier::low_order(conduit::Node &dataset)
 #endif
 }
 
+bool Transmogrifier::is_poly(const conduit::Node &n)
+{
+  conduit::Node actual = n.children().next();
```
Comment (JustinPrivitera) [inline src/ascent/runtimes/ascent_transmogrifier.cpp]: Is there a cleaner way to do this?

File: src/ascent/runtimes/ascent_transmogrifier.cpp [hunk 2]
```diff
@@ -107,6 +109,147 @@ conduit::Node* Transmogrifier::low_order(conduit::Node &dataset)
...
+    }
+  }
+
+  return false;
+}
+
+// to_poly assumes that the node n is polyhedral
+conduit::Node* Transmogrifier::to_poly(conduit::Node &n)
+{
+  conduit::Node actual = n.children().next();
```
Comment (JustinPrivitera) [inline src/ascent/runtimes/ascent_transmogrifier.cpp]: Same thing here

File: src/ascent/runtimes/ascent_transmogrifier.cpp [hunk 3]
```diff
@@ -107,6 +109,149 @@ conduit::Node* Transmogrifier::low_order(conduit::Node &dataset)
...
+        res["topologies/" + poly_topos[i]],
+        res["coordsets/" + coordsets[coordsets.size() - 1]],
+        res["fields"],
+        s2dmap,
+        d2smap,
+        options);
+    }
+
+    // TODO want to set_external FIRST and then change stuff w/ gen_sides
+    // maybe delete the stuff that is going to get overwritten in the dest
```
Comment (JustinPrivitera) [inline src/ascent/runtimes/ascent_transmogrifier.cpp]: This is what I will work on next

File: src/ascent/runtimes/ascent_transmogrifier.cpp [hunk 4]
```diff
@@ -107,6 +109,151 @@ conduit::Node* Transmogrifier::low_order(conduit::Node &dataset)
...
+          if (topo["elements/shape"].as_string() == "polyhedral" || 
+              topo["elements/shape"].as_string() == "polygonal")
+          {
+            poly_topos.push_back(topo.name());
+          }
+        }
+      }
+    }
+
+    res.set_external(dom);
```
Comment (JustinPrivitera) [inline src/ascent/runtimes/ascent_transmogrifier.cpp:169]: All the tests work without me having to delete the parts of the nodes that are being overwritten; should I not worry about it then?

File: src/ascent/runtimes/ascent_data_object.cpp [hunk 5]
```diff
@@ -305,11 +305,29 @@ std::shared_ptr<VTKHCollection> DataObject::as_vtkh_collection()
...
+      {
+        Transmogrifier::to_poly(*m_low_bp, *to_vtkh);
+        zero_copy = false;
+      }
+      else
+      {
+        to_vtkh->set_external(*m_low_bp);
+      }
+    }
+
```
Comment (cyrush) [inline src/ascent/runtimes/ascent_data_object.cpp:324]:
```
    bool zero_copy = true;

    conduit::Node  n_poly;
    conduit::Node *to_vtkh = nullptr;

    if (m_low_bp != nullptr)
    {
      if (Transmogrifier::is_poly(*m_low_bp))
      {
        Transmogrifier::to_poly(*m_low_bp, n_poly);
        to_vtkh = &n_poly;
        zero_copy = false;
      }
      else
      {
        to_vtkh = m_low_bp;
      }
    }

    // convert to vtkh
    std::shared_ptr<VTKHCollection>
      vtkh_dset(VTKHDataAdapter::BlueprintToVTKHCollection(*to_vtkh, zero_copy));

    m_vtkh = vtkh_dset;
    
    return m_vtkh;
```

File: src/ascent/runtimes/ascent_transmogrifier.cpp (hunk 4 above)
Comment (cyrush) [reply src/ascent/runtimes/ascent_transmogrifier.cpp:169]: it should magically work :-)

**Pull request merged**